src/main/java/dev/walshy/sfmetrics/MetricsModule.java

```diff
@@ -1,47 +1,77 @@
 
     private MetricsModule() {}
 
-    @SuppressWarnings("unused")
     public static void start() {
-        Metrics metrics = new Metrics(SlimefunPlugin.instance(), 4574);
+        Metrics metrics = new Metrics(SlimefunPlugin.instance(), PLUGIN_ID);
+        branch = SlimefunPlugin.getUpdater().getBranch();
+        slimefunVersion = SlimefunPlugin.getUpdater().getBuildNumber();
 
-        if (SlimefunPlugin.getUpdater().getBranch().isOfficial()) {
-            // We really do not need this data if it is an unofficially modified build...
-            metrics.addCustomChart(new AutoUpdaterChart());
-        }
+        addChart(metrics, AutoUpdaterChart::new);
+        addChart(metrics, ResourcePackChart::new);
+        addChart(metrics, SlimefunVersionChart::new);
+        addChart(metrics, ServerLanguageChart::new);
+        addChart(metrics, PlayerLanguageChart::new);
+        addChart(metrics, ResearchesEnabledChart::new);
+        addChart(metrics, GuideLayoutChart::new);
+        addChart(metrics, AddonsChart::new);
+        addChart(metrics, CommandChart::new);
+        addChart(metrics, ServerSizeChart::new);
+        addChart(metrics, CompatibilityModeChart::new);
+        addChart(metrics, MetricsVersionChart::new);
+        addChart(metrics, NewServersChart::new);
+        addChart(metrics, MetricsAutoUpdatesChart::new);
 
-        metrics.addCustomChart(new ResourcePackChart());
-        metrics.addCustomChart(new SlimefunVersionChart());
-        metrics.addCustomChart(new ServerLanguageChart());
-        metrics.addCustomChart(new PlayerLanguageChart());
-        metrics.addCustomChart(new ResearchesEnabledChart());
-        metrics.addCustomChart(new GuideLayoutChart());
-        metrics.addCustomChart(new AddonsChart());
-        metrics.addCustomChart(new CommandChart());
-        metrics.addCustomChart(new ServerSizeChart());
-        metrics.addCustomChart(new CompatibilityModeChart());
-        metrics.addCustomChart(new MetricsVersionChart());
-
-        SlimefunPlugin.instance().getLogger().info("Now running MetricsModule v" + VERSION);
+        SlimefunPlugin.instance().getLogger().log(Level.INFO, "Now running MetricsModule build #{0}", VERSION);
+        SlimefunPlugin.instance().getLogger().log(Level.INFO, "with a total of {0} chart(s)!", charts);
+    }
+
+    private static void addChart(Metrics metrics, Supplier<CustomChart> constructor) {
+        try {
+            CustomChart chart = constructor.get();
+
+            if (chart instanceof VersionDependentChart && !((VersionDependentChart) chart).isCompatible(branch, slimefunVersion)) {
+                // Not compatible with this Slimefun version
+                return;
+            }
+
+            metrics.addCustomChart(chart);
+            charts++;
+        }
+        catch (Exception | LinkageError x) {
+            SlimefunPlugin.instance().getLogger().log(Level.WARNING, x, () -> "Failed to load a bStats chart for Metrics #" + VERSION);
```

Comment:
We should post the class name here so we know which chart.

Reply:
Was my initial thought too, but then again: The stacktrace would show it already since each chart has its own class.

Reply:
Not sure we should send a stacktrace and not just log name. I've tried to not display anything in this module as an error as it doesn't actually affect SF. I don't want people coming and saying "SF is broken" because we made a typo in the version let's say.

Reply:
Well, to be fair this try/catch should never be executed anyway.
The actual methods are only called when bStats decides to transmit data.
This try/catch only really errors when we screwed up massively.